tests/integration_test.rs

```diff
@@ -2,104 +2,117 @@ extern crate sonos;
+async fn get_speaker() -> sonos::Speaker {
+    let devices = sonos::discover().await.unwrap();
 
     devices
         .into_iter()
-        .find(|d| d.name == "Bedroom")
+        .find(|d| d.name == "Living Room")
         .ok_or("Couldn't find bedroom")
         .unwrap()
 }
 
-#[test]
-fn can_discover_devices() {
-    let devices = sonos::discover().unwrap();
+#[tokio::test]
+async fn can_discover_devices() {
+    let devices = sonos::discover().await.unwrap();
     assert!(!devices.is_empty(), "No devices discovered");
 }
 
-#[test]
-fn volume() {
-    let device = get_speaker();
-    device.set_volume(2).expect("Failed to get volume");
+#[tokio::test]
+async fn volume() {
+    let device = get_speaker().await;
+    device.set_volume(2).await.expect("Failed to get volume");
     assert_eq!(
-        device.volume().expect("Failed to get volume"),
+        device.volume().await.expect("Failed to get volume"),
         2 as u8,
         "Volume was not updated."
     );
 }
 
-#[test]
-fn muted() {
-    let device = get_speaker();
-    device.mute().expect("Couldn't mute player");
+#[tokio::test]
+async fn muted() {
+    let device = get_speaker().await;
+    device.mute().await.expect("Couldn't mute player");
     assert_eq!(
-        device.muted().expect("Failed to get current mute status"),
+        device
+            .muted()
+            .await
+            .expect("Failed to get current mute status"),
         true
     );
-    device.unmute().expect("Couldn't unmute player");
+    device.unmute().await.expect("Couldn't unmute player");
     assert_eq!(
-        device.muted().expect("Failed to get current mute status"),
+        device
+            .muted()
+            .await
+            .expect("Failed to get current mute status"),
         false
     );
 }
 
-#[test]
-fn playback_state() {
-    let device = get_speaker();
+#[tokio::test]
+async fn playback_state() {
+    let device = get_speaker().await;
 
-    device.play().expect("Couldn't play track");
-    assert!(match device.transport_state().unwrap() {
+    device.play().await.expect("Couldn't play track");
+    assert!(match device.transport_state().await.unwrap() {
         TransportState::Playing | TransportState::Transitioning => true,
         _ => false,
     });
 
-    device.pause().expect("Couldn't pause track");
-    assert!(match device.transport_state().unwrap() {
+    device.pause().await.expect("Couldn't pause track");
+    assert!(match device.transport_state().await.unwrap() {
         TransportState::PausedPlayback | TransportState::Transitioning => true,
         _ => false,
     });
 
-    device.stop().expect("Couldn't stop track");
-    assert!(match device.transport_state().unwrap() {
+    device.stop().await.expect("Couldn't stop track");
+    let state = device.transport_state().await.unwrap();
+    // eprintln!("{:#?}", state);
+    // This returns PausedPlayback on my speaker - is stop no longer supported?
```

Comment:
@w4 See comment

Reply:
This library hasn't been updated since S1 was around, I'll see if I can find some time to fix everything broken with the move to S3 & clean up the API a little.